microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/MicroProfileDocumentHighlight.java

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lsp4mp.services;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.lsp4j.DocumentHighlight;
+import org.eclipse.lsp4j.DocumentHighlightKind;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4mp.ls.commons.BadLocationException;
+import org.eclipse.lsp4mp.model.Node;
+import org.eclipse.lsp4mp.model.PropertiesModel;
+import org.eclipse.lsp4mp.model.Property;
+import org.eclipse.lsp4mp.model.PropertyKey;
+import org.eclipse.lsp4mp.model.PropertyValueExpression;
+import org.eclipse.lsp4mp.model.Node.NodeType;
+import org.eclipse.lsp4mp.utils.PositionUtils;
+
+/**
+ * Highlight for micro profile config documents
+ */
+public class MicroProfileDocumentHighlight {
+
+	/**
+	 * Returns a list of highlights for a MicroProfile properties document
+	 *
+	 * @param document The MicroProfile properties document
+	 * @param position The position of the cursor
+	 * @return a list of highlights for a MicroProfile properties document
+	 */
+	public List<? extends DocumentHighlight> doDocumentHighlight(PropertiesModel document, Position position) {
```

Comment:
I will rename to `findDocumentHighlights` to be consistent with XML Language Service, and TypeScript vscode-html language service, json, etc